frontend/src/lib/ConnectionManager.ts

```diff
@@ -25,9 +25,12 @@ import { ensureError } from "./ErrorHandling"
 
 /**
  * When the websocket connection retries this many times, we show a dialog
- * letting the user know we're having problems connecting.
+ * letting the user know we're having problems connecting. This happens
+ * after about 15 seconds since on the 6th retry, we've set timeouts for
+ * a total of approximately 0.5 + 1 + 2 + 4 + 8 = 15.5 seconds (+/- some
+ * due to jitter).
  */
-const RETRY_COUNT_FOR_WARNING = 30 // around 15s
+const RETRY_COUNT_FOR_WARNING = 6
```

Comment:
Just check do we have off by one error?
is 6 mean: `0.5 + 1 + 2 + 4 + 8 + 16 = ` `31` seconds, or I missed the first initial connection attempt?

Reply:
Ah, the dialog is displayed before the 6th retry, so there's no +16. Let me tweak the wording in the comment to make this more clear.